pom.xml

```diff
@@ -64,6 +64,11 @@
         <sonar.host.url>https://builds.apache.org/analysis</sonar.host.url>
         <!-- Exclude all generated code -->
         <sonar.exclusions>**/generated-sources</sonar.exclusions>
+        <!-- whether using a thrift compiler locally
+        (if false, maven will download a binary thrift compiler).
+        if true, you must set the thrift compiler into your PATH, and make sure
+        the version is ${thrift.version}-->
+        <use.local.thrift.compiler>false</use.local.thrift.compiler>
```

Comment:
if i use `mvn clean compile -Duse.local.thrift.compiler=true` command, both `thrift-download-and-generation` and `thrift-generation-using-local-compiler` will be triggered?

Reply:
Um... I am trying to avoid it .
It is strange, Though I set use.local.thrift.compiler=false by default, but if I add a activation condition here:
                <property>
                    <name>use.local.thrift.compiler</name>
                    <value>false</value>
                </property> 

We have to claim the value: mvn clean compile -Duse.local.thrift.compiler=false
I do not know why ..